doc/adr/0010-leaf-cluster-k8s-auth.md

```diff
@@ -0,0 +1,63 @@
+# 10. Leaf Cluster Kubernetes API Authentication
+
+Date: 2021-11-16
+
+## Status
+
+Accepted
+
+## Context
+
+When a user "onboards" a new Leaf Cluster, we need ensure that the Weave GitOps API deployment running on the Management Cluster can query data about the objects on the Leaf Cluster. At present, we would have to rely on the user's local `kubeconfig` to provide all necessary authentication credentials to query multiple clusters at once.
+
+This approach won't work for most use cases, as we cannot expect every user of the WeGO UI (and subsequently API) to have access or authorization to view objects on all clusters.
+
+## Design
+
+To allow a user to query data about a Leaf Cluster, we must do (roughly) these steps during the "onboarding" CLI command:
+
+1. Create a `ServiceAccount` on the Leaf Cluster with cluster admin priveleges
+2. Add the `kubeconfig` for that `ServiceAccount` as a `Secret` on the Management Cluster.
+3. Add a `Cluster` Custom Resource to the Management Cluster to persist the `Cluster` record for future retreival
```

Comment:
If I'm reading this correctly, the `Cluster` CR is added to the management cluster, before it's been approved via PR?

How do you detect rejection of the PR and remove the secret and `Cluster`?

Reply:
That's a valid question.  Doesn't this issue extend to the current implementation of `gitops install` in general?  If you run the gitops install it will apply the flux CRDs and everything else to the cluster today.  The user would have to run `gitops uninstall` in that scenario to remove everything that was applied to the cluster.